libr/parse/p/parse_x86_pseudo.c

```diff
@@ -358,12 +364,58 @@ static inline void mk_reg_str(const char *regname, int delta, bool sign, bool at
 	r_strbuf_free (sb);
 }
 
-static bool varsub (RParse *p, RAnalFunction *f, ut64 addr, int oplen, char *data, char *str, int len) {
-	RList *bpargs, *spargs;
+static const RList *__args(RParse *p, RAnalFunction *f, int type) {
+#if CACHE_VARS
```

Comment:
if CACHE_VARS brings any substantial improvement, please just keep the CACHE_VARS version and remove the rest of the code before PR's approval.